packages/core/src/utils/setup-package-json.ts

```diff
@@ -85,18 +88,18 @@ export const diffPackageJson = (
 
   const packageJsonExports = packageJson.exports;
 
-  if (!isDeepStrictEqual(packageJsonExports, exportsKey)) {
+  if (!isDeepStrictEqual(packageJsonExports, exports)) {
     diffs.push({ key: 'exports' });
   }
 
   const expectedPackageJson =
     diffs.length > 0
-      ? {
+      ? ({
           main,
           module,
           files: filesArray.sort((a, b) => (a > b ? 1 : -1)),
-          exports: exportsKey,
-        }
+          exports,
+        } as PackageJson)
```

Comment:
Would it be better to add an explict type annotation to `expectedPackageJson` rather than casting here?

Reply:
Already done:

https://github.com/seek-oss/crackle/blob/74b8d934c6e0670abb6f0794210218fce446826a/packages/core/src/utils/setup-package-json.ts#L40-L46

But because `exports`'s type is `` `./${string}` ``, we have to cast.